source/kraken/tests/realtime_test.cpp

```diff
@@ -2545,3 +2570,64 @@ BOOST_AUTO_TEST_CASE(add_new_with_earlier_arrival_and_delete_existingstop_time_i
     BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(3).is_detour(), false);
     BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(3).departure_status(), pbnavitia::StopTimeUpdateStatus::UNCHANGED);
 }
+
+BOOST_AUTO_TEST_CASE(should_get_base_stoptime_with_realtime_added_stop_time) {
+    ed::builder b("20171101");
+
+    b.sa("A");
+    b.sa("B");
+    b.sa("C");
+    b.sa("D");
+
+    b.vj("L1").uri("vj:1")
+        ("stop_point:A", "08:10"_t)
+        ("stop_point:C", "08:30"_t)
+        ("stop_point:D", "08:40"_t);
```

Comment:
It would be nice to add a stop at C or B at the end, make also a delay message on all stops, and see how it behaves also on a trip from A to C and from D to C (only not crashing is OK ;-) )

Reply:
The original problem was that the arrivals/departures **after** an added stop time were wrong. I don"t see how adding a stop at the end would help in this regard. 

Handling lollipops is another matter that should be fixed separately as it already (partially) broken.